### PR TITLE
Add Flexibility to Commands AddToRecipe and DeleteFromRecipe

### PR DESCRIPTION
**Pull request by YatPang**

### CODE REVIEW

File: src/main/java/seedu/duke/command/Command.java [hunk 1]
```diff
@@ -86,6 +89,77 @@ public void execute(UI ui) throws IOException {
...
+                    break;
+                }
+                if (id.trim().equals(StringLib.EMPTY_STRING) && elementType.equals("i")) {
+                    ui.showEmptyIngredientID();
+                    break;
+                }
+                Recipe recipeToAddTo = RecipeList.viewRecipe(id);
+                int index;
+                switch (elementType) {
+                case "s":
```
Comment (liuziyang020319) [inline src/main/java/seedu/duke/command/Command.java:128]: avoid magic string

File: src/main/java/seedu/duke/parser/Parser.java [hunk 2]
```diff
@@ -283,10 +289,109 @@ public static void parseEditStep(Integer recipeIndex, String description) throws
...
+        return count;
+    }
+    public static boolean isValidAddToRecipe(String description) {
+        String descLowerCase = description.toLowerCase().trim();
+        if (!(descLowerCase.contains("--s") && descLowerCase.contains("--i")) &&
+                matchCount(descLowerCase,"id/") == 1 &&
+                matchCount(descLowerCase,"desc/") == 1 &&
+                (matchCount(descLowerCase, "--i") == 1 ||
+                        matchCount(descLowerCase, "--s") == 1) &&
+                !(descLowerCase.contains("--is") ||
```
Comment (liuziyang020319) [inline src/main/java/seedu/duke/parser/Parser.java]: if you do want to accept --iXX just remove this judge

File: docs/UserGuide.md [hunk 3]
```diff
@@ -84,6 +86,35 @@ Now you have 2 recipes in the list.
 Data saved successfully!
 __________________________________________________________
 ```
+<h2 id="adding-elements-to-recipe"> Adding an element to a recipe: <code class="language-plaintext highlighter-rouge">addtorecipe</code></h2>
+
+Adds a recipe to the recipe manager. App will prompt you to start entering the steps to the recipe will automatically count the steps and add the recipe to the recipe list.<br>
+**Format**: `addtorecipe --[s/i] id/[index] desc/[description of step/ingredient]`<br>
+**Hint**: Order of `--[s/i] id/[index] desc/[description of step/ingredient]` is flexible<br>
```
Comment (liuziyang020319) [inline docs/UserGuide.md:93]: told the user that --iXXX is also ok

**Review by LimHongYao — APPROVED** [review]
+737 lines ez
Will raise issue on checking for default case for all switch-case and if-else-if-else (Code Quality)

**Pull request merged**